null cannot be cast to non-null type org.jetbrains.uast.UParameter in `PermissionsLaunchDetector`

## Describe the bug

We started receiving this issue in our CI environment after upgrading dependencies (100% repro in my local environment as well). From what I can see, it seems like a bug in the `com.google.accompanist.permissions.lint.PermissionsLaunchDetector` detector.

```
* What went wrong:
Execution failed for task ':app:lintAnalyzeAppDebug'.
> A failure occurred while executing com.android.build.gradle.internal.lint.AndroidLintWorkAction
   > Unexpected failure during lint analysis (this is a bug in lint or one of the libraries it depends on)
     
     Message: Unexpected failure during lint analysis (this is a bug in lint or one of the libraries it depends on)
     
     Message: Unexpected failure during lint analysis of Permissions.kt (this is a bug in lint or one of the libraries it depends on)
     
     Message: null cannot be cast to non-null type org.jetbrains.uast.UParameter
     
     The crash seems to involve the detector \\\`com.google.accompanist.permissions.lint.PermissionsLaunchDetector\\\`.
     You can try disabling it with something like this:
         android {
             lint {
                 disable "PermissionLaunchedDuringComposition"
             }
         }
```

<details><summary>Full stack trace</summary>

(Stack traces have been prettified for readability)

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:lintAnalyzeAppDebug'.
> A failure occurred while executing com.android.build.gradle.internal.lint.AndroidLintWorkAction
   > Unexpected failure during lint analysis (this is a bug in lint or one of the libraries it depends on)
     
     Message: Unexpected failure during lint analysis (this is a bug in lint or one of the libraries it depends on)
     
     Message: Unexpected failure during lint analysis of Permissions.kt (this is a bug in lint or one of the libraries it depends on)
     
     Message: null cannot be cast to non-null type org.jetbrains.uast.UParameter
     
     The crash seems to involve the detector \\\`com.google.accompanist.permissions.lint.PermissionsLaunchDetector\\\`.
     You can try disabling it with something like this:
         android {
             lint {
                 disable "PermissionLaunchedDuringComposition"
             }
         }

Stack:
  NullPointerException:
    ComposableUtilsKt.isComposable(ComposableUtils.kt:124)
    ←ComposableUtilsKt.isComposable(ComposableUtils.kt:135)
    ←ComposableBodyVisitor.isComposable(ComposableUtils.kt:161)
    ←ComposableUtilsKt.isInvokedWithinComposable(ComposableUtils.kt:59)
    ←PermissionsLaunchDetector.visitMethodCall(PermissionsLaunchDetector.kt:52)
    ←UElementVisitor$DelegatingUastVisitor.visitMethodCallExpression(UElementVisitor.kt:780)
    ←UElementVisitor$DelegatingUastVisitor.visitCallExpression(UElementVisitor.kt:761)
    ←UCallExpression.accept(UCallExpression.kt:94)
    ←UQualifiedReferenceExpression.accept(UQualifiedReferenceExpression.kt:34)
    ←UReturnExpression.accept(UReturnExpression.kt:22)
    ←ImplementationUtilsKt.acceptList(implementationUtils.kt:15)
    ←UBlockExpression.accept(UBlockExpression.kt:21)
    ←ULambdaExpression.accept(ULambdaExpression.kt:40)
    ←ImplementationUtilsKt.acceptList(implementationUtils.kt:15)
    ←UCallExpression.accept(UCallExpression.kt:98)
    ←UReturnExpression.accept(UReturnExpression.kt:22)
    ←ImplementationUtilsKt.acceptList(implementationUtils.kt:15)
    ←UBlockExpression.accept(UBlockExpression.kt:21)
    ←ULambdaExpression.accept(ULambdaExpression.kt:40)
    ←ImplementationUtilsKt.acceptList(implementationUtils.kt:15)
    ←UCallExpression.accept(UCallExpression.kt:98)
    ←ImplementationUtilsKt.acceptList(implementationUtils.kt:15)
    ←UBlockExpression.accept(UBlockExpression.kt:21)
    ←UMethod.accept(UMethod.kt:45)
    ←ImplementationUtilsKt.acceptList(implementationUtils.kt:15)
    ←AbstractKotlinUClass.accept(AbstractKotlinUClass.kt:213)
    ←ImplementationUtilsKt.acceptList(implementationUtils.kt:15)
    ←UFile.accept(UFile.kt:89)
    ←UastLintUtilsKt.acceptSourceFile(UastLintUtils.kt:953)
    ←UElementVisitor$visitFile$3.run(UElementVisitor.kt:216)
    ←LintClient.runReadAction(LintClient.kt:1781)
    ←LintDriver$LintClientWrapper.runReadAction(LintDriver.kt:2747)
    ←UElementVisitor.visitFile(UElementVisitor.kt:213)
    ←LintDriver$visitUastDetectors$1.run(LintDriver.kt:2048)
    ←LintClient.runReadAction(LintClient.kt:1781)
    ←LintDriver$LintClientWrapper.runReadAction(LintDriver.kt:2747)
    ←LintDriver.visitUastDetectors(LintDriver.kt:2048)
    ←LintDriver.visitUast(LintDriver.kt:2010)
    ←LintDriver.runFileDetectors(LintDriver.kt:1284)
    ←LintDriver.checkProject(LintDriver.kt:1070)
    ←LintDriver.checkProjectRoot(LintDriver.kt:622)
    ←LintDriver.access$checkProjectRoot(LintDriver.kt:174)
    ←LintDriver$analyzeOnly$1.invoke(LintDriver.kt:445)
    ←LintDriver$analyzeOnly$1.invoke(LintDriver.kt:442)
    ←LintDriver.doAnalyze(LintDriver.kt:501)
    ←LintDriver.analyzeOnly(LintDriver.kt:442)
    ←LintCliClient$analyzeOnly$1.invoke(LintCliClient.kt:258)
    ←LintCliClient$analyzeOnly$1.invoke(LintCliClient.kt:255)
    ←LintCliClient.run(LintCliClient.kt:312)
    ←LintCliClient.analyzeOnly(LintCliClient.kt:255)
    ←Main.run(Main.java:1766)
    ←Main.run(Main.java:282)
    ←DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    ←Method.invoke(Method.java:578)
    ←AndroidLintWorkAction.invokeLintMainRunMethod(AndroidLintWorkAction.kt:103)
    ←AndroidLintWorkAction.runLint(AndroidLintWorkAction.kt:90)
    ←AndroidLintWorkAction.execute(AndroidLintWorkAction.kt:64)
    ←DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
    ←NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:66)
    ←NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:62)
    ←ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
    ←NoIsolationWorkerFactory$1.lambda$execute$0(NoIsolationWorkerFactory.java:62)
    ←AbstractWorker$1.call(AbstractWorker.java:44)
    ←AbstractWorker$1.call(AbstractWorker.java:41)
    ←DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
    ←DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
    ←DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
    ←DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
    ←DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
    ←DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
    ←DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
    ←AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
    ←NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:59)
    ←DefaultWorkerExecutor.lambda$submitWork$0(DefaultWorkerExecutor.java:174)
    ←FutureTask.run(FutureTask.java:317)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:195)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.access$700(DefaultConditionalExecutionQueue.java:128)
    ←DefaultConditionalExecutionQueue$ExecutionRunner$1.run(DefaultConditionalExecutionQueue.java:170)
    ←Factories$1.create(Factories.java:31)
    ←DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:267)
    ←DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:131)
    ←DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:136)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:165)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:134)
    ←Executors$RunnableAdapter.call(Executors.java:577)
    ←FutureTask.run(FutureTask.java:317)
    ←ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
    ←AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
    ←ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    ←ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    ←Thread.run(Thread.java:1589)

Stack:
  RuntimeException:
    LintDriver$Companion.handleDetectorError(LintDriver.kt:3740)
    ←LintDriver$Companion.handleDetectorError$default(LintDriver.kt:3607)
    ←LintDriver$Companion.handleDetectorError(LintDriver.kt:3603)
    ←UElementVisitor.visitFile(UElementVisitor.kt:242)
    ←LintDriver$visitUastDetectors$1.run(LintDriver.kt:2048)
    ←LintClient.runReadAction(LintClient.kt:1781)
    ←LintDriver$LintClientWrapper.runReadAction(LintDriver.kt:2747)
    ←LintDriver.visitUastDetectors(LintDriver.kt:2048)
    ←LintDriver.visitUast(LintDriver.kt:2010)
    ←LintDriver.runFileDetectors(LintDriver.kt:1284)
    ←LintDriver.checkProject(LintDriver.kt:1070)
    ←LintDriver.checkProjectRoot(LintDriver.kt:622)
    ←LintDriver.access$checkProjectRoot(LintDriver.kt:174)
    ←LintDriver$analyzeOnly$1.invoke(LintDriver.kt:445)
    ←LintDriver$analyzeOnly$1.invoke(LintDriver.kt:442)
    ←LintDriver.doAnalyze(LintDriver.kt:501)
    ←LintDriver.analyzeOnly(LintDriver.kt:442)
    ←LintCliClient$analyzeOnly$1.invoke(LintCliClient.kt:258)
    ←LintCliClient$analyzeOnly$1.invoke(LintCliClient.kt:255)
    ←LintCliClient.run(LintCliClient.kt:312)
    ←LintCliClient.analyzeOnly(LintCliClient.kt:255)
    ←Main.run(Main.java:1766)
    ←Main.run(Main.java:282)
    ←DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    ←Method.invoke(Method.java:578)
    ←AndroidLintWorkAction.invokeLintMainRunMethod(AndroidLintWorkAction.kt:103)
    ←AndroidLintWorkAction.runLint(AndroidLintWorkAction.kt:90)
    ←AndroidLintWorkAction.execute(AndroidLintWorkAction.kt:64)
    ←DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
    ←NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:66)
    ←NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:62)
    ←ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
    ←NoIsolationWorkerFactory$1.lambda$execute$0(NoIsolationWorkerFactory.java:62)
    ←AbstractWorker$1.call(AbstractWorker.java:44)
    ←AbstractWorker$1.call(AbstractWorker.java:41)
    ←DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
    ←DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
    ←DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
    ←DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
    ←DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
    ←DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
    ←DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
    ←AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
    ←NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:59)
    ←DefaultWorkerExecutor.lambda$submitWork$0(DefaultWorkerExecutor.java:174)
    ←FutureTask.run(FutureTask.java:317)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:195)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.access$700(DefaultConditionalExecutionQueue.java:128)
    ←DefaultConditionalExecutionQueue$ExecutionRunner$1.run(DefaultConditionalExecutionQueue.java:170)
    ←Factories$1.create(Factories.java:31)
    ←DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:267)
    ←DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:131)
    ←DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:136)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:165)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:134)
    ←Executors$RunnableAdapter.call(Executors.java:577)
    ←FutureTask.run(FutureTask.java:317)
    ←ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
    ←AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
    ←ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    ←ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    ←Thread.run(Thread.java:1589)

Stack:
  RuntimeException:
    LintDriver$Companion.handleDetectorError(LintDriver.kt:3740)
    ←LintDriver$Companion.handleDetectorError$default(LintDriver.kt:3607)
    ←LintDriver$Companion.handleDetectorError(LintDriver.kt:3603)
    ←LintDriver$analyzeOnly$1.invoke(LintDriver.kt:447)
    ←LintDriver$analyzeOnly$1.invoke(LintDriver.kt:442)
    ←LintDriver.doAnalyze(LintDriver.kt:501)
    ←LintDriver.analyzeOnly(LintDriver.kt:442)
    ←LintCliClient$analyzeOnly$1.invoke(LintCliClient.kt:258)
    ←LintCliClient$analyzeOnly$1.invoke(LintCliClient.kt:255)
    ←LintCliClient.run(LintCliClient.kt:312)
    ←LintCliClient.analyzeOnly(LintCliClient.kt:255)
    ←Main.run(Main.java:1766)
    ←Main.run(Main.java:282)
    ←DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    ←Method.invoke(Method.java:578)
    ←AndroidLintWorkAction.invokeLintMainRunMethod(AndroidLintWorkAction.kt:103)
    ←AndroidLintWorkAction.runLint(AndroidLintWorkAction.kt:90)
    ←AndroidLintWorkAction.execute(AndroidLintWorkAction.kt:64)
    ←DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
    ←NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:66)
    ←NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:62)
    ←ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
    ←NoIsolationWorkerFactory$1.lambda$execute$0(NoIsolationWorkerFactory.java:62)
    ←AbstractWorker$1.call(AbstractWorker.java:44)
    ←AbstractWorker$1.call(AbstractWorker.java:41)
    ←DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
    ←DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
    ←DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
    ←DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
    ←DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
    ←DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
    ←DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
    ←AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
    ←NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:59)
    ←DefaultWorkerExecutor.lambda$submitWork$0(DefaultWorkerExecutor.java:174)
    ←FutureTask.run(FutureTask.java:317)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:195)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.access$700(DefaultConditionalExecutionQueue.java:128)
    ←DefaultConditionalExecutionQueue$ExecutionRunner$1.run(DefaultConditionalExecutionQueue.java:170)
    ←Factories$1.create(Factories.java:31)
    ←DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:267)
    ←DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:131)
    ←DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:136)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:165)
    ←DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:134)
    ←Executors$RunnableAdapter.call(Executors.java:577)
    ←FutureTask.run(FutureTask.java:317)
    ←ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
    ←AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
    ←ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    ←ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    ←Thread.run(Thread.java:1589)

```
</details>

## To Reproduce

Sorry, no repro at this time.

```
$ ./gradlew :app:lintAnalyzeAppDebug
```

## Expected behavior
No crash - build should succeed.

## Environment:
 - GitHub Actions
 - Accompanist version: `com.google.accompanist:accompanist-permissions:0.32.0`

```
$ ./gradlew --version

------------------------------------------------------------
Gradle 8.9
------------------------------------------------------------

Build time:    2024-07-11 14:37:41 UTC
Revision:      d536ef36a19186ccc596d8817123e5445f30fef8

Kotlin:        1.9.23
Groovy:        3.0.21
Ant:           Apache Ant(TM) version 1.10.13 compiled on January 4 2023
Launcher JVM:  19.0.2 (Private Build 19.0.2+7-Ubuntu-0ubuntu322.04)
Daemon JVM:    /usr/lib/jvm/java-19-openjdk-amd64 (no JDK specified, using current Java home)
OS:            Linux 6.8.0-49-generic amd64
```

--- a/gradle.properties
+++ b/gradle.properties
@@ -33,7 +33,7 @@ systemProp.org.gradle.internal.http.socketTimeout=120000
 
 GROUP=com.google.accompanist
 # !! No longer need to update this manually when using a Compose SNAPSHOT
-VERSION_NAME=0.36.1-SNAPSHOT
+VERSION_NAME=0.37.0
 
 POM_DESCRIPTION=Utilities for Jetpack Compose
 

